www_src/pages/login/sign-up.jsx

```diff
@@ -0,0 +1,162 @@
+var React = require('react/addons');
+var api = require('../../lib/api');
+var validationMixin = require('../../lib/validators');
+var FormInput = require('./form-input.jsx');
+
+// <SignUp />
+// Component for the Sign Up/Create user form. See Login view for usage.
+var SignUp = React.createClass({
+
+  mixins: [React.addons.LinkedStateMixin, validationMixin],
+
+  // Props:
+  //   show
+  //     boolean
+  //     Shows the entire component if true, hides if false
+  //   setParentState
+  //     function(state)
+  //     Calls setState on parent component (Login) given a state object.
+  //     Useful for setting the loading or mode state of the Login view.
+  //
+  getDefaultProps: function () {
+    return {};
+  },
+
+  // State:
+  //   globalError
+  //     string or boolean
+  //     Indicates whether or not a an error was received from the server.
+  //   username
+  //   email
+  //   password
+  //     string
+  //     These store state for the form input fields
+  //   feedback
+  //     boolean
+  //     Stores state for feedback checkbox
+
+  getInitialState: function () {
+    return {
+      globalError: false,
+      feedback: false
+    };
+  },
+
+  // Consumed by validationMixin
+  // and used to generate form fields
+  fields: [
+    {
+      name: 'username',
+      label: 'Username',
+      required: true,
+      validations: 'username'
+    },
+    {
+      name: 'email',
+      label: 'Email',
+      type: 'email',
+      required: true,
+      validations: 'email'
+    },
+    {
+      name: 'password',
+      label: 'Password',
+      type: 'password',
+      required: true,
+      validations: ['passwordLength', 'lowerCase', 'upperCase', 'numbers']
+    }
+  ],
+
+  // Called when the form is submitted
+  // Triggers an api call via api.signUp
+  // If an error is received, sets state.globalError
+  // If success:
+  //    1. resets form
+  //    2. caches user session in Android
+  //    3. redirects to /main
+  onSubmit: function (e) {
+    e.preventDefault();
+
+    var errors = this.getValidationErrors();
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+
+    var json = {
```

Comment:
Could we call this `options`? I generally find `json` to be a little confusing for me whenever I run into it and it isn't a serialized string.